src/stores/linkStore.ts

```diff
@@ -27,9 +35,34 @@ export class LinkStore {
     @action
     public setLink = (link: ILink) => {
         this._link = link;
+        this._undoObjects = [{
+            link,
+        }];
+        this._undoIndex = 0;
         this.setOldLink(link);
     }
 
+    @action
+    public updateLinkGeometry = (latLngs: L.LatLng[]) => {
+        if (!this._link) return;
+
+        const updatedLink = _.cloneDeep(this._link);
+        updatedLink.geometry = latLngs;
+
+        const currentUndoObject = {
```

Comment:
Would suggest typing to check that object complies with what we are creating.
const currentUndoObject: UndoObject = {
}

Reply:
:+1: